src/main/java/com/powsybl/openloadflow/sensi/DcSensitivityAnalysis.java

```diff
@@ -608,20 +756,62 @@ protected DenseMatrix initContingencyRhs(LfNetwork lfNetwork, EquationSystem equ
 
     private void detectConnectivityLoss(LfNetwork lfNetwork, DenseMatrix states, List<Contingency> contingencies, Map<String, ComputedContingencyElement> contingenciesElements,
                                         EquationSystem equationSystem, Collection<Contingency> nonLosingConnectivityContingencies,
-                                        Map<List<Set<ComputedContingencyElement>>, List<Contingency>> contingenciesByGroupOfElementsBreakingConnectivity) {
+                                        Map<Set<ComputedContingencyElement>, List<Contingency>> contingenciesByGroupOfElementsBreakingConnectivity) {
         for (Contingency contingency : contingencies) {
-            List<Set<ComputedContingencyElement>> groupOfElementsBreakingConnectivity = getGroupOfElementsBreakingConnectivity(lfNetwork, states,
+            Set<ComputedContingencyElement> groupOfElementsBreakingConnectivity = getGroupOfElementsBreakingConnectivity(lfNetwork, states,
                     contingency.getElements().stream().map(element -> contingenciesElements.get(element.getId())).collect(Collectors.toList()), equationSystem);
             if (groupOfElementsBreakingConnectivity.isEmpty()) { // connectivity not broken
                 nonLosingConnectivityContingencies.add(contingency);
-            } else if (groupOfElementsBreakingConnectivity.stream().mapToInt(Set::size).max().getAsInt() > 1) {
-                throw new PowsyblException("The contingency " + contingency.getId() + " breaks the connectivity on more than one branch.");
             } else {
                 contingenciesByGroupOfElementsBreakingConnectivity.computeIfAbsent(groupOfElementsBreakingConnectivity, key -> new LinkedList<>()).add(contingency);
```

Comment:
From another PR, but a `Set.put()` would be clearer (always absent).

Reply:
It shouldn't always be absent
The very purpose of this function is to index contingencies by group of elements that breaks connectivity.
For example, if you have two contingencies removing the following lines: {L1, L2}, and {L1, L3}, and the line L1 breaks the connectivity: then groupOfElementsBreakingConnectivity should be {L1} for both contingencies

Reply:
Thanks for explaining. My mistake was due to the fact that I had in mind that Set.equals() was not checking more than the object equality, but in fact it's doing more than that. A test with the example you're describing would have avoided my mistake! :) But again, this is not the purpose of this PR anyway.